### PR TITLE
Update to 3rdparty v82

### PR DESCRIPTION
**Pull request by rh101**
## Describe your changes

Tested with the latest OpenSSL changes, and everything seems to be working correctly.

### For each PR
- [ ] Add Copyright if it missed:   
      - `"Copyright (c) 2019-present Axmol Engine contributors (see AUTHORS.md)."`
- [ ] I have performed a self-review of my code.
       
   Optional:
   - [ ] I have checked readme and add important infos to this PR.
   - [ ] I have added/adapted some tests too.
          
### For core/new feature PR
- [ ] I have checked readme and add important infos to this PR.
- [ ] I have added thorough tests.

### CODE REVIEW

File: 3rdparty/CMakeLists.txt [hunk 1]
```diff
@@ -306,7 +306,9 @@ if(ANDROID OR LINUX)
     target_link_libraries(OpenSSL::SSL INTERFACE OpenSSL::Crypto)
     target_link_libraries(3rdparty INTERFACE OpenSSL::SSL)
 elseif(APPLE)
-    target_link_libraries(3rdparty INTERFACE openssl)
+    target_link_libraries(3rdparty
```
Comment (halx99) [inline 3rdparty/CMakeLists.txt]: Use alias targets same with other platforms?

Comment (halx99) [reply 3rdparty/CMakeLists.txt]: Better solution:
```cmake
# -----macro: set openssl sub target-----
  macro(set_openssl_sub_target sub_target_name sub_lib_name)
    
    if(WINDOWS)
      add_library(${sub_target_name} SHARED IMPORTED GLOBAL)
    else()
      add_library(${sub_target_name} STATIC IMPORTED GLOBAL)
    endif()

    set_target_properties(${sub_target_name} PROPERTIES
      INTERFACE_INCLUDE_DIRECTORIES "${openssl_INC_DIR}"
    )

    if(WINDOWS)
      set(impl_lib_suffix "")
      if(WIN64)
          set(impl_lib_suffix "-${ARCH_ALIAS}")
      endif()
      set_target_properties(${sub_target_name} PROPERTIES
        IMPORTED_LOCATION "${openssl_LIB_DIR}/lib${sub_lib_name}-3${impl_lib_suffix}.dll"
        IMPORTED_IMPLIB "${openssl_LIB_DIR}/lib${sub_lib_name}.lib"
      )
    elseif(APPLE)
       set_target_properties(${sub_target_name} PROPERTIES
        IMPORTED_LOCATION "${openssl_LIB_DIR}/ossl-${sub_lib_name}.xcframework"
      )
    else()
      set_target_properties(${sub_target_name} PROPERTIES
        IMPORTED_LOCATION "${openssl_LIB_DIR}/lib${sub_lib_name}.a"
      )
    endif()
    if(${sub_lib_name} STREQUAL "ssl")
      set(OPENSSL_ROOT_DIR "${openssl_LIB_DIR}" CACHE STRING "" FORCE)
      set(OPENSSL_SSL_LIBRARY "OpenSSL::SSL" CACHE STRING "" FORCE)
    else()
      set(OPENSSL_CRYPTO_LIBRARY "OpenSSL::Crypto" CACHE STRING "" FORCE)
    endif()
  endmacro()
```

Comment (halx99) [reply 3rdparty/CMakeLists.txt]: And 3rdparty/CMakeLists.txt:
```cmake
add_subdirectory(openssl)
if(ANDROID OR LINUX)
    target_link_libraries(OpenSSL::SSL INTERFACE OpenSSL::Crypto)
    target_link_libraries(3rdparty INTERFACE OpenSSL::SSL)
else()
    target_link_libraries(3rdparty
                        INTERFACE OpenSSL::Crypto
                        INTERFACE OpenSSL::SSL)
endif()
```

Comment (rh101) [reply 3rdparty/CMakeLists.txt]: That is better. I'll update it now.

**Review by halx99 — APPROVED** [review]

**Pull request merged**